### PR TITLE
Proxy forms for http/https proxies

### PR DESCRIPTION
**Pull request by ibolton336**

### CODE REVIEW

**Review by gildub — COMMENTED** [review]
Looks good so far.

File: pkg/client/src/app/pages/proxies/proxy-form.tsx [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+      [IDENTITY]: isHttpIdentityRequired || isHttpsIdentityRequired,
+    }),
+    onSubmit: onSubmit,
+  });
+
+  const onChangeIsHttpIdentityRequired = () => {
+    setIsHttpIdentityRequired(!isHttpIdentityRequired);
+  };
+
+  const onChangeIsHttpsIdentityRequired = () => {
```
Comment (gildub) [inline pkg/client/src/app/pages/proxies/proxy-form.tsx:142]: Not used

File: pkg/client/src/app/pages/proxies/proxy-form.tsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+  isSecure?: boolean;
+  onSaved: (response: AxiosResponse<any>) => void;
+  onCancel: () => void;
+}
+
+export const ProxyForm: React.FC<ProxyFormProps> = ({
+  proxy,
+  isSecure,
+  onSaved,
+  onCancel,
```
Comment (gildub) [inline pkg/client/src/app/pages/proxies/proxy-form.tsx:55]: Not used

**Review by gildub — COMMENTED** [review]
Looks good.
Just need a cleanup.
There are some unused functions

**Review by gildub — COMMENTED** [review]
When reloading the proxy page, after setting up one of the proxies, the switch button is showing deactivated.

The proxy page also needs to have the Textbox for the list of hosts to exclude from HTTP/HTTPS proxy.

File: pkg/client/src/app/pages/proxies/proxies.tsx [hunk 3]
```diff
@@ -11,18 +11,60 @@ import {
...
-    setHttpProxy(!isHttpProxy);
+  // const onChangeIsHttpProxy = () => {
+  //   setIsHttpProxy(!isHttpProxy);
+  //   fetchProxies();
+  // };
+
+  // const onChangeIsHttpsProxy = () => {
+  //   setIsHttpsProxy(!isHttpsProxy);
+  //   fetchProxies();
+  // };
```
Comment (gildub) [inline pkg/client/src/app/pages/proxies/proxies.tsx:33]: Do we need to keep this for later ? If yes then ignore my comment.

File: pkg/client/src/app/pages/proxies/proxies.tsx [hunk 4]
```diff
@@ -11,18 +11,60 @@ import {
...
+  const existingHttpsProxy = proxies?.data.find(
+    (proxy) => proxy.kind === "https"
+  );
+  // const [isHttpProxy, setIsHttpProxy] = React.useState(false);
+  // const [isHttpsProxy, setIsHttpsProxy] = React.useState(false);
+
+  // useEffect(() => {
+  //   setIsHttpProxy(!!existingHttpProxy);
+  //   setIsHttpsProxy(!!existingHttpsProxy);
+  // }, []);
```
Comment (gildub) [inline pkg/client/src/app/pages/proxies/proxies.tsx:65]: same here ?

File: pkg/client/src/app/pages/proxies/proxies-validation-schema.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    [HTTPS_PORT]: yup.lazy(() =>
+      mandatoryFields.httpPort
+        ? yup.string().required(REQUIRED_MESSAGE)
+        : yup.string()
+    ),
+    [HTTPS_HOST]: yup.lazy(() =>
+      mandatoryFields[HTTPS_HOST]
+        ? yup.string().required(REQUIRED_MESSAGE)
+        : yup.string()
+    ),
```
Comment (gildub) [inline pkg/client/src/app/pages/proxies/proxies-validation-schema.ts:70]: This is already defined above.

**Review by gildub — CHANGES_REQUESTED** [review]
I wasn't able to save a proxy configuration.
Also the exclude list is not specific to https proxy but both

**Review by gildub — COMMENTED** [review]
Code looks good.
Meanwhile when I test none of the Proxy switches are enabling.

**Comment by ibolton336** [comment]
> Code looks good. Meanwhile when I test none of the Proxy switches are enabling.

Do you have the latest images? @gildub

**Review by gildub — CHANGES_REQUESTED** [review]
Just tested against latest hub.
I can save fine but I can't disable after enabling either http or https proxy.
Also noted that the exclude list needs to be separate from proxies.

**Review by gildub — APPROVED** [review]
LGTM

**Pull request merged**